physionet-django/project/models.py

```diff
@@ -1533,28 +1533,179 @@ def get_storage_info(self, force_calculate=True):
-                doi=self.doi)
-        else:
-            return format_html(
-                '{authors} ({year}). {title} (version {version}). '
-                '<i>PhysioNet</i>.',
-                authors=', '.join(a.initialed_name() for a in authors),
-                year=self.publish_datetime.year,
-                title=self.title,
-                version=self.version)
+        citation_dict = {}
+
+        for style in styles:
+
+            if (style == 'MLA'):
+
+                if (len(authors) == 1):
+                    temp_authors = authors[0].get_full_name(reverse=True)
+                elif (len(authors) == 2):
+                    first_author = authors[0].get_full_name(reverse=True)
+                    second_author = authors[1].get_full_name()
+                    temp_authors = first_author + ', and ' + second_author
+                else:
+                    temp_authors = ', '.join(authors[0].get_full_name().split())
+                    temp_authors += ', et al'
+
+                if self.doi:
+                    citation_format = format_html(
+                        '{authors}. "{title} (version {version})." '
+                        '<i>PhysioNet</i> ({year}), doi:'
+                        '<a href="https://doi.org/{doi}">https://doi.org/{doi}</a>.',
+                        authors=temp_authors,
+                        title=self.title,
+                        version=self.version,
+                        year=self.publish_datetime.year,
+                        doi=self.doi)
+                else:
+                    citation_format = format_html(
+                        '{authors}. "{title} (version {version})." '
+                        '<i>PhysioNet</i> ({year}).',
+                        authors=temp_authors,
+                        title=self.title,
+                        version=self.version,
+                        year=self.publish_datetime.year)
+
+            elif (style == 'APA'):
+
+                if (len(authors) == 1):
+                    temp_authors = authors[0].initialed_name()
+                elif (len(authors) == 2):
+                    first_author = authors[0].initialed_name()
+                    second_author = authors[1].initialed_name()
+                    temp_authors = first_author + ', & ' + second_author
+                elif (len(authors) > 20):
+                    temp_authors = ', '.join(a.initialed_name() for a in authors[0:19])
+                    temp_authors += ',... ' + authors[len(authors)-1].initialed_name()
+                else:
+                    temp_authors = ', '.join(a.initialed_name() for a in authors[:(len(authors)-1)])
+                    temp_authors += ', & ' + authors[len(authors)-1].initialed_name()
+
+                if self.doi:
+                    citation_format = format_html(
+                        '{authors} ({year}). {title} (version {version}). '
+                        '<i>PhysioNet</i>. '
```

Comment:
Ideally we should avoid hardcoding `PhysioNet` here. It would be good if it could be taken from an environment variable.

Reply:
Will this ever change? Maybe it's in a fixture somewhere?

Reply:
The idea is for the platform software to be reusable, so it may be different for other people. It probably needs to be added as a new variable in the .env file (oddly, I don't think it's defined at the moment!). Maybe REPOSITORY_NAME or something?

Reply:
Let's make this another pull request?

Reply:
Yep, another PR would be fine.